examples/update-on-async-rendering/react-dom-properties-before-update-after.js

```diff
@@ -0,0 +1,35 @@
+class ScrollingList extends React.Component {
+  listRef = null;
+
+  // highlight-range{1-8}
+  getSnapshotBeforeUpdate(prevProps, prevState) {
+    // Are we adding new items to the list?
+    // Capture the current height of the list so we can adjust scroll later.
+    if (prevProps.list.length < this.props.list.length) {
+      return this.listRef.scrollHeight;
+    }
+    return null;
+  }
+
+  // highlight-range{1-8}
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    // If we have a snapshot value, we've just added new items.
```

Comment:
Maybe add a comment:

// `snapshot` here will be the value returned from getSnapshotBeforeUpdate.